### PR TITLE
Pre-requesites to test discovery V7

### PR DESCRIPTION
**Pull request by lmr**
Preliminary commits to the latest version of the test discovery PR, that will succeed #358.

Just one fix and one utility function.

### CODE REVIEW

File: avocado/utils/astring.py [hunk 1]
```diff
@@ -114,3 +114,46 @@ def strip_console_codes(output, custom_codes=None):
...
+def matrix_to_string(matrix, header=None):
+    """
+    Return a pretty, aligned string representation of a nxm matrix.
+
+    This representation can be used to print any tabular data, such as
+    database results. It works by scanning the lengths of each element
+    in each column, and determining the format string dynamically.
+
+    :param matrix: Matrix representation (list with n rows of m elements).
+    :param header: Optional tuple or list with header elements to be displayed.
```
Comment (ruda) [inline avocado/utils/astring.py]: Optional sequence with header elements to be displayed.

File: avocado/utils/astring.py [hunk 2]
```diff
@@ -114,3 +114,46 @@ def strip_console_codes(output, custom_codes=None):
         old_word = tmp_word
         return_str += tmp_word[len(special_code):]
     return return_str
+
+
+def matrix_to_string(matrix, header=None):
```
Comment (ruda) [inline avocado/utils/astring.py]: Why not `tabular_data()`?

Comment (lmr) [inline avocado/utils/astring.py]: It's a decent name. I'll change it.

**Comment by lmr** [comment]
@ruda, done with the changes.

**Comment by ruda** [comment]
LGTM.

**Pull request merged**